lib/rack/lint.rb

```diff
@@ -655,6 +657,29 @@ def check_hijack_response(headers, env)
     ## * Middleware should not wrap the IO object for the request pattern. The
     ##   request pattern is intended to provide the hijacker with "raw tcp".
 
+    ## === Early Hints
+    ##
+    ## The application or any middleware may call the <tt>rack.early_hints</tt>
+    ## with pairs of headers that will be sent as a 103 Early Hints provisional
+    ## response.
+    def check_early_hints(env)
+      if env[RACK_EARLY_HINTS]
+        ##
+        ## If rack.early_hints is present it must respond to #call.
+        assert("rack.early_hints must respond to call") { env[RACK_EARLY_HINTS].respond_to?(:call) }
+        original_callback = env[RACK_EARLY_HINTS]
+        env[RACK_EARLY_HINTS] = lambda do |headers|
+          ## The headers must respond to #each_pair
+          assert("rack.early_hints headers must respond to each_pair") { headers.respond_to?(:each_pair) }
+          headers.each_pair do |key, value|
+            ## The headers keys and values must respond to #to_s
```

Comment:
Period at the end of sentences in Lint comments, since they are combined in SPEC.  Otherwise I think RDoc makes them appear as a single sentence.

It would also be better to be more descriptive. `The rack.early_hints argument must respond to #each_pair and yield a key string and a value string to the provided block for each header.`  This would change the specification slightly (`is_a?(String)` instead of `respond_to?(:to_s)`).